### PR TITLE
feat: add deploy pepeline command with required and optional flags

### PR DESCRIPTION
**Pull request by EstebanRomero84**
### What does this PR do?
Adds new command **sf deploy pipeline**. Only includes the required and optional flags as well as its validations.

### What issues does this PR fix or reference?

[W-11912468](https://gus.lightning.force.com/lightning/a07EE000018q8J3YAI)
[W-11991482](https://gus.lightning.force.com/lightning/a07EE00001BF8ADYA1)
[W-11991498](https://gus.lightning.force.com/lightning/a07EE00001BFISPYA5)

### CODE REVIEW

File: src/commands/deploy/pipeline.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  devopsCenterProjectName,
+  devopsCenterUsername,
+  specificTests,
+  testLevel,
+} from '../../common/flags';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-devops-center', 'deploy.pipeline');
+
+export default class DeployPipeline extends SfCommand<PromotePipelineResult> {
```
Comment (JuanStenghele-sf) [inline src/commands/deploy/pipeline.ts:24]: Just a small detail here, in [W-11991482](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001BF8ADYA1/view) we name this class Pipeline.

File: command-snapshot.json [hunk 2]
```diff
@@ -1 +1,17 @@
-[]
+[
+  {
+    "command": "deploy:pipeline",
+    "plugin": "@salesforce/plugin-devops-center",
+    "flags": [
+      "branch-name",
+      "bundle-version-name",
```
Comment (JuanStenghele-sf) [inline command-snapshot.json:7]: I could not find this command in any of the WI but I think it is OK as it is in the docs.

File: messages/commonFlags.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# promote.devops-center-project-name.summary
+
+Name of the DevOps Center project.
+
+# promote.branch-name.summary
+
+Name of the branch from which to deploy changes to the stage’s org.
```
Comment (JuanStenghele-sf) [inline messages/commonFlags.md]: This is slightly different from the docs: "Name of the branch in the source control repository from which to deploy changes to the stage’s org."

File: messages/commonFlags.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+Username or alias for the DevOps Center org.
+
+# promote.bundle-version-name.summary
+
+Version name of the bundle.
+
+# promote.bundle-version-name.description
+
+Bundle version name when deploying to the first stage after the bundling stage.
```
Comment (JuanStenghele-sf) [inline messages/commonFlags.md]: Found this in the docs: "You must indicate the bundle version if deploying to the environment that correspond to the first stage after the bundling stage."

File: messages/deploy.pipeline.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+  <%= config.bin %> <%= command.id %> --devops-center-project-name “Recruiting App” --branch-name staging --devops-center-username MyStagingSandbox --bundle-version-name 1.0
+
+- Deploy all changes in the main branch to the release environment:
+
+  <%= config.bin %> <%= command.id %> --devops-center-project-name “Recruiting App” --branch-name main --devops-center-username MyReleaseOrg --deploy-all
+
+# error.BranchNotFound
+
+Can't find any branch named %s in project %s.
```
Comment (JuanStenghele-sf) [inline messages/deploy.pipeline.md:21]: I could not find this in the docs. Did Emily check this?

File: messages/deploy.pipeline.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+# error.NoTestsSpecified
+
+You must specify tests using the --tests flag if the --test-level flag is set to RunSpecifiedTests.
```
Comment (JuanStenghele-sf) [inline messages/deploy.pipeline.md:25]: I could not find this in the docs. Did Emily check this?

File: messages/deploy.pipeline.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+# error.InvalidRunTests
+
+runTests can only be used with a testLevel of RunSpecifiedTests.
```
Comment (JuanStenghele-sf) [inline messages/deploy.pipeline.md:29]: I could not find this in the docs. Did Emily check this?

File: messages/commonFlags.md (hunk 3 above)
Comment (EstebanRomero84) [reply messages/commonFlags.md]: Fixed, thank you.

File: messages/commonFlags.md (hunk 4 above)
Comment (EstebanRomero84) [reply messages/commonFlags.md]: Fixed

File: messages/deploy.pipeline.md (hunk 6 above)
Comment (EstebanRomero84) [reply messages/deploy.pipeline.md:25]: Emily and I agreed that for flags and/or validations that are the same than in `sf deploy metadata` standard command we should use the same message. In this case I got it from [here](https://github.com/salesforcecli/plugin-deploy-retrieve/blob/main/messages/deploy.metadata.md#:~:text=You%20must%20specify%20tests%20using%20the%20%2D%2Dtests%20flag%20if%20the%20%2D%2Dtest%2Dlevel%20flag%20is%20set%20to%20RunSpecifiedTests.).

File: messages/deploy.pipeline.md (hunk 7 above)
Comment (EstebanRomero84) [reply messages/deploy.pipeline.md:29]: Same here, I got it from the standard command.

File: src/commands/deploy/pipeline.ts (hunk 1 above)
Comment (EstebanRomero84) [reply src/commands/deploy/pipeline.ts:24]: I thought it would be better to have a more specific class name.

Comment (JuanStenghele-sf) [reply src/commands/deploy/pipeline.ts:24]: :nice:

File: messages/deploy.pipeline.md (hunk 6 above)
Comment (JuanStenghele-sf) [reply messages/deploy.pipeline.md:25]: Ok!

File: test/commands/deploy/pipeline.test.ts [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { expect, test } from '@oclif/test';
+
+describe('validate flags', () => {
+  test
+    .stdout()
```
Comment (JuanStenghele-sf) [inline test/commands/deploy/pipeline.test.ts:12]: I think we do not need this line, as we only check stderr

File: test/commands/deploy/pipeline.test.ts [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    .stderr()
+    .command(['deploy:pipeline', '--branch-name=test'])
+    .it('runs deploy pipeline with no provided project name', (ctx) => {
+      expect(ctx.stderr).to.contain('Missing required flag devops-center-project-name');
+    });
+
+  test
+    .stdout()
```
Comment (JuanStenghele-sf) [inline test/commands/deploy/pipeline.test.ts:20]: I think we do not need this line, as we only check stderr

File: test/commands/deploy/pipeline.test.ts [hunk 10]
```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { expect, test } from '@oclif/test';
+
+describe('validate flags', () => {
```
Comment (JuanStenghele-sf) [inline test/commands/deploy/pipeline.test.ts:10]: These tests are ok. I thought of adding new tests with the happy path but as the command is not fully implemented yet we may leave this for the future. What do you think?

**Comment by JuanStenghele-sf** [comment]
Nice job!

File: test/commands/deploy/pipeline.test.ts (hunk 10 above)
Comment (EstebanRomero84) [reply test/commands/deploy/pipeline.test.ts:10]: Test for happy path added, thx

**Review by JuanStenghele-sf — APPROVED** [review]
Nice job!

**Pull request merged**